posts/2023-08-22-23.0.0.8.adoc

```diff
@@ -0,0 +1,288 @@
+
+
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/25739
+// Contact/Reviewer: ayoho
+// // // // // // // // 
+[#pkce]
+== Prevent authorization code interception attacks with PKCE support for OpenID Connect clients
+OpenID Connect clients in Liberty now support Proof Key for Code Exchange (PKCE) (link:https://datatracker.ietf.org/doc/html/rfc7636[RFC 7636]). PKCE is an extension of the OAuth 2.0 specification and provides protection from authorization code interception attacks for OAuth 2.0 public clients. In very specific scenarios, a malicious application can intercept an authorization code intended for a legitimate OAuth 2.0 public client and use the authorization code to obtain access and ID tokens on behalf of the client. PKCE introduces additional steps and request parameters to prevent such interception attacks.
+
+Enable this functionality using the `pkceCodeChallengeMethod` attribute in either the `<openidConnectClient>` or `<oidcLogin>` elements in the `server.xml`.
+
+For example, when you use the OpenID Connect Client feature, include configuration similar to the following example:
+
+[source,xml]
+----
+    <featureManager>
+        <feature>openidConnectClient-1.0</feature>
+    </featureManager>
+    ...
+    <openidConnectClient pkceCodeChallengeMethod="S256" ... />
+----
+
+If you are using the Social Media Login feature, include configuration similar to the following example:
+
+[source,xml]
+----
+    <featureManager>
+        <feature>socialLogin-1.0</feature>
+    </featureManager>
+    ...
+    <oidcLogin pkceCodeChallengeMethod="S256" ... />
+----
+
+For more information about the configuration options, refer to the docs for the link:{url-prefix}/docs/latest/reference/config/openidConnectClient.html[openidConnectClient element] and the link:{url-prefix}/docs/latest/reference/config/oidcLogin.html[oidcLogin element].
+
+   
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 
+
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/25926
+// Contact/Reviewer: Azquelt
+// // // // // // // // 
+[#featureinstall]
+== Ensure sufficient features are installed when using `featureUtility installFeature` command
+When `featureUtility installFeature <featurename>` is used to install a feature on the command line, the feature and all required dependencies are installed.
+
+However, this doesn't guarantee that the feature will start correctly when used with other features in the server, particularly for features that can work with multiple versions of other features. This means that you could list all the features you wanted to use on the command line but find that they didn't all work together because `featureUtility` hadn't installed the right version of every dependency.
```

Comment:
thanks for this explanation. I kept tripping over this sentence because you have a feature that _can_ work with multiple versions resulting in a dependency conflict, which is somewhat counterintuitive. But I understand why now and I can't think of a way to make it more plain without getting unnecessarily deep in the weeds.